packages/react-native-bridge/android/react-native-bridge/build.gradle

```diff
@@ -8,8 +10,23 @@ plugins {
 // import the `readReactNativeVersion()` function
 apply from: 'https://gist.githubusercontent.com/hypest/742448b9588b3a0aa580a5e80ae95bdf/raw/8eb62d40ee7a5104d2fcaeff21ce6f29bd93b054/readReactNativeVersion.gradle'
 
-// import the `readHashedVersion()` function
-apply from: 'https://gist.githubusercontent.com/hypest/ceaf20a8e7d9b8404e4a5ff2e6c36650/raw/e1460a128e4b9863963410d719c7d44c3adefd02/readHashedVersion.gradle'
+def readHashedVersion(packagejson, packageName, section) {
+    def packageSlurper = new JsonSlurper()
+    def packageJson = packageSlurper.parse file(packagejson)
+    def dep = packageJson.get(section).get(packageName)
+    
+    // Extract version from tarball URL
+    def isTarball = dep.endsWith('.tgz')
+    if(isTarball) {
+        def fileName = dep.substring(dep.lastIndexOf('/') + 1)
+        def version = fileName.replace(packageName + '-', '').replace('.tgz', '')
+        return version
+    }
+
+    // Extract version from Git URL
+    def hash = dep.substring(dep.lastIndexOf('#') + 1)
+    return hash;
+}
```

Comment:
@hypest I've copy/pasted the function you have in a Gist, I'd appreciate it if you could take a look at the new changes and let me know if you have any feedback 🙇 .

Reply:
Thanks for the ping Carlos!

Looks good to me in terms of the code to extract the hash/tag 👍

Only one comment/question I guess. Feels like the "hashed" naming might not be accurate anymore? Looks like we are interested in the tag instead. Do we expect cases that it will be a git hash proper? If only expecting tags now, perhaps renaming to `readPackageTag` is a better name to convey the purpose of the function/operation? And if so, renaming the local vars accordingly too. WDYT?

Reply:
Agreed, I'm thinking to rename it to something like `extractPackageVersion`. I'd like to avoid including in the name a reference to the part that we want to extract from the package URL, as this function could handle multiple. Instead, we could use what we're expecting to extract, in this case, the version of the package.

> Do we expect cases that it will be a git hash proper?
Not initially as we're going to use the tarball URL for all packages but I decided to leave the option if in the future we revert some of them to Git URLs.

Reply:
Agreed 👍. I see you made the change in 52193b7dffa24b42ae5dd613c5b8b288b1530f07 so, I'll mark this thread as resolved.